core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterComponentUpdateService.java

```diff
@@ -0,0 +1,78 @@
+package com.sequenceiq.cloudbreak.service;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.model.component.PreparedImages;
+import com.sequenceiq.cloudbreak.cluster.service.ClusterComponentConfigProvider;
+import com.sequenceiq.cloudbreak.common.json.Json;
+import com.sequenceiq.cloudbreak.common.type.ComponentType;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.Cluster;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.ClusterComponent;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterService;
+
+@Component
+public class ClusterComponentUpdateService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClusterComponentUpdateService.class);
+
+    @Inject
+    private ClusterService clusterService;
+
+    @Inject
+    private ClusterComponentConfigProvider clusterComponentConfigProvider;
+
+    public void updateOrSavePreparedClusterComponent(long stackId, ComponentType componentType,
```

Comment:
I think `ComponentType componentType` parameter should be removed, as this method works only with this type of component only

Reply:
done